service-base/src/main/java/org/eclipse/hono/service/monitoring/ConnectionEventProducer.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+package org.eclipse.hono.service.monitoring;
+
+import org.eclipse.hono.service.auth.device.Device;
+
+/**
+ * Produces connection events.
+ * <p>
+ * The interface is intended to be implemented by setups which are interested in receiving technical connection events.
+ * Which might not make sense for all protocols adapters. But mostly for connection oriented ones like MQTT or AMQP 1.0.
+ * <p>
+ * The protocol adapters may call {@link #connected(String, Device)} and {@link #disconnected(String, Device)} as they
+ * see fit. The whole process is a "best effort" process and intended to be used for monitoring/debugging.
+ * <p>
+ * When a protocol adapter calls into this producer it must provide some kind of connection ID, which might have a
+ * different meaning for different protocol adapters. This should be as identifiable as possible but there is no
+ * requirement for this to be unique. However for each connection the protocol adapter must call
+ * {@link #disconnected(String, Device)} with the same information as it called {@link #connected(String, Device)}.
+ */
+public interface ConnectionEventProducer {
+
+    /**
+     * Produce an event for a new connection.
+     * 
+     * @param remoteId The ID of the remote endpoint which connected (e.g. a remote address, port, client id, ...).
+     * @param authenticatedDevice The optional authenticated device associated with the connection. May be {@code null}
+     *            if the connection is from an unauthenticated device.
+     */
+    public void connected(String remoteId, Device authenticatedDevice);
```

Comment:
no need to declare public visibility, this is an interface

Reply:
what is the value of *remoteId* and *authenticatedDevice* in case of an device that has not been authenticated?

Reply:
What value does this have then? You do not know which device this is because within Hono the device is identified, well, by the device-id (which is **not** the remoteId, is it?).

Reply:
Well with MQTT CONNECT you would still present some ID. Exactly the remote id is not the device id. AMQP would still maybe present come container/link ID. Or we could use the remote endpoint address (IP+Port) or IP+Port+ClientID for MQTT.

The idea is to have some kind of protocol adapter specific ID, which is available even without having an authenticated Hono device. As this is for monitoring/debugging purposes, something which helps you find out "who this is".